.github/workflows/azure-e2e-test.yml

```diff
@@ -0,0 +1,177 @@
+name: azure-e2e-test
+
+env:
+  PODVM_GALLERY_NAME: "podvm_gallery0"
+  PODVM_IMAGE_DEF_NAME: "podvm_image0"
+  PODVM_IMAGE_VERSION: "0.0.1"
+  PODVM_IMAGE_NAME: "peerpod-image-0.0.1"
+  SSH_USERNAME: "peerpod"
+  VM_SIZE: "Standard_D2as_v5"
+
+on:
+  push:
+    branches:
+      - kartikjoshi21/ci-upstream
+
+jobs:
+  build-caa-image:
+    runs-on: ubuntu-latest
+    environment: build
+    steps:
+    - uses: actions/checkout@v2
+
+    - uses: ./.github/actions/login-to-azure
+      with:
+        client-id: ${{ secrets.AZURE_CLIENT_ID }}
+        client-secret: ${{ secrets.AZURE_CLIENT_SECRET }}
+        subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+        tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+
+    - name: Docker login
+      run: |
+        TOKEN=$(az acr login --name ${{ secrets.ACR_URL }} --expose-token --output tsv --query accessToken)
+        docker login ${{ secrets.ACR_URL }} --username 00000000-0000-0000-0000-000000000000 --password "$TOKEN"
+
+    - name: Build and push CAA docker image
+      env:
+        registry: ${{ secrets.ACR_URL }}
+        CLOUD_PROVIDER: azure
+      run: |
+        docker buildx create --name multiarch --driver docker-container --use
+        make image
+
+  build-podvm-image:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+
+    - uses: ./.github/actions/login-to-azure
+      with:
+        client-id: ${{ secrets.AZURE_CLIENT_ID }}
+        client-secret: ${{ secrets.AZURE_CLIENT_SECRET }}
+        subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+        tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+
+    - uses: ./.github/actions/setup-azure-build-env
+
+    - name: Clone kata repository
+      run: |
+        pushd ../
+        git clone https://github.com/kata-containers/kata-containers -b CCv0
+        popd  
+
+    - name: Create podvm image
+      id: create-podvm-image
+      env:
+        PKR_VAR_subscription_id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+        PKR_VAR_client_id: ${{ secrets.AZURE_CLIENT_ID }}
+        PKR_VAR_client_secret: ${{ secrets.AZURE_CLIENT_SECRET }}
+        PKR_VAR_tenant_id: ${{ secrets.AZURE_TENANT_ID }}
+        PKR_VAR_resource_group: ${{ secrets.AZURE_RESOURCE_GROUP }}
+        PKR_VAR_location: ${{ secrets.AZURE_REGION }}
+        PKR_VAR_az_image_name: ${{ env.PODVM_IMAGE_NAME }}
+        PKR_VAR_vm_size: ${{ env.VM_SIZE }}
+        PKR_VAR_ssh_username: ${{ env.SSH_USERNAME }}
+        PKR_VAR_az_gallery_name: ${{ env.PODVM_GALLERY_NAME }}
+        PKR_VAR_az_gallery_image_name: ${{ env.PODVM_IMAGE_DEF_NAME }}
+        PKR_VAR_az_gallery_image_version: ${{ env.PODVM_IMAGE_VERSION }}
```

Comment:
Since those env's are used only in this step, can we just set them here? like:

`PKR_VAR_vm_size: "Standard_D2as_v5"`

Reply:
Later on we can plan to provide them as part of secrets.